sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -428,6 +430,25 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
     plan.children.foreach(assertIsOnTheGpu(_, conf))
   }
 
+  /**
+   * This is intended for testing only and this only supports looking for an exec once.
+   */
+  private def validateExecsInGpuPlan(plan: SparkPlan, conf: RapidsConf): Unit = {
+    val validateExecs = conf.validateExecsInGpuPlan.toSet
+    if (validateExecs.nonEmpty) {
```

Comment:
I think the check is redundant and the logic inside still works on the empty set or is it for performance?

Reply:
yes it was meant to be short cut to skip below logic altogether.